x-pack/legacy/plugins/apm/index.ts

```diff
@@ -77,7 +77,10 @@ export const apm: LegacyPluginInitializer = kibana => {
         autocreateApmIndexPattern: Joi.boolean().default(true),
 
         // service map
-        serviceMapEnabled: Joi.boolean().default(true)
+        serviceMapEnabled: Joi.boolean().default(true),
+
+        // telemetry
+        telemetryCollectionEnabled: Joi.boolean().default(true)
```

Comment:
do we want a specific flag to disable APM-only telemetry (while the rest of the telemetry is enabled)? Wouldn't that complicate the analysis about the usage? aka: how can I tell APM is used but not reporting telemetry because it's disabled? 
We could do that by checking the `application_usage.apm` stats... but I'd like @alexfrancoeur thoughts on this.

Reply:
Would be great to get some additional feedback 👍🏻the reason why I've added it because we want an escape hatch in case the queries are too expensive, without requiring our users to disable telemetry altogether in those cases.

Reply:
@afharo If you don't mind, I'll merge this in given the time constraints, but let me know if there's anything we need to change.